api/src/test/java/uk/gov/dhsc/htbhf/claimant/message/processor/DetermineEntitlementMessageProcessorTest.java

```diff
@@ -146,16 +224,162 @@ void shouldUpdateClaimWhenClaimantIsNotEligible(EligibilityStatus eligibilitySta
+        verifyZeroInteractions(childDateOfBirthCalculator);
+    }
+
+    //HTBHF-1757 Claimant had children in previous cycle who would be under 4 in this cycle but are not returned from DWP
+    @Test
+    void shouldUpdateClaimToPendingExpiryAndSendNoLongerEligibleEmailWhenClaimantIsNotEligibleWithNoChildrenInCurrentCycle() {
+        //Given
+        List<LocalDate> previousCycleChildrenDobs = TWO_CHILDREN;
+        List<LocalDate> currentCycleChildrenDobs = NO_CHILDREN;
+        LocalDate currentCycleExpectedDeliveryDate = NOT_PREGNANT;
+        LocalDate previousCycleExpectedDeliveryDate = NOT_PREGNANT;
+        DetermineEntitlementMessageContext context = buildMessageContext(
+                previousCycleExpectedDeliveryDate,
+                currentCycleExpectedDeliveryDate,
+                previousCycleChildrenDobs,
+                currentCycleChildrenDobs);
+        given(messageContextLoader.loadDetermineEntitlementContext(any())).willReturn(context);
+        given(pregnancyEntitlementCalculator.isEntitledToVoucher(any(), any())).willReturn(false);
+        //Had children in last cycle who are under 4 at the start of the current payment cycle.
+        given(childDateOfBirthCalculator.hadChildrenUnder4AtStartOfPaymentCycle(any())).willReturn(true);
+        given(childDateOfBirthCalculator.hadChildrenUnderFourAtGivenDate(any(), any())).willReturn(true);
+
+        //Eligibility - any status that isn't ELIGIBLE has the same effect here.
+        EligibilityAndEntitlementDecision decision = aDecisionWithStatusAndChildren(INELIGIBLE, NOT_CONFIRMED, currentCycleChildrenDobs);
+        given(eligibilityAndEntitlementService.evaluateExistingClaimant(any(), any(), any())).willReturn(decision);
+
+        //Current payment cycle voucher entitlement mocking
+        Message message = aValidMessageWithType(DETERMINE_ENTITLEMENT);
+
+        //When
+        MessageStatus messageStatus = processor.processMessage(message);
+
+        //Then
+        runCommonVerifications(context, decision, message, messageStatus, ClaimStatus.PENDING_EXPIRY);
         verify(determineEntitlementNotificationHandler).sendClaimNoLongerEligibleEmail(context.getClaim());
-        verifyZeroInteractions(pregnancyEntitlementCalculator);
+        LocalDate currentPaymentCycleStartDate = context.getCurrentPaymentCycle().getCycleStartDate();
+        verify(pregnancyEntitlementCalculator).isEntitledToVoucher(NOT_PREGNANT, currentPaymentCycleStartDate);
+        verify(childDateOfBirthCalculator).hadChildrenUnder4AtStartOfPaymentCycle(context.getPreviousPaymentCycle());
+        verify(childDateOfBirthCalculator).hadChildrenUnderFourAtGivenDate(TWO_CHILDREN, currentPaymentCycleStartDate);
+        verifyZeroInteractions(childDateOfBirthCalculator);
+    }
+
+    @ParameterizedTest(name = "Qualifying benefit status={0}")
+    @ValueSource(strings = {"CONFIRMED", "NOT_CONFIRMED"})
+    void shouldExpireClaimWhenClaimantWasPregnantButNoLongerWithChildrenInPreviousCycle(
```

Comment:
I wanted to combine this test and the next test but they would need conditional logic on the mocks so decided to separate them out into two tests, if anyone can see a good way, please let me know.